addons/sale/views/sale_order_line_views.xml

```diff
@@ -25,32 +25,76 @@
         <field name="name">sale.order.line.form.readonly</field>
         <field name="model">sale.order.line</field>
         <field name="arch" type="xml">
-            <form string="Sales Order Item">
+            <form string="Sales Order Item" edit="false">
+                <field name="display_type" invisible="1"/>
                 <sheet>
                     <div class="oe_title">
                         <h1>
-                            <field name="display_name" readonly="1"/>
+                            <field name="display_name"/>
                         </h1>
                     </div>
                     <group>
                         <group>
+                            <field name="product_updatable" invisible="1"/>
                             <field name="order_id" readonly="1"/>
-                            <field name="product_id" readonly="1"/>
-                            <field name="name" readonly="1"/>
-                            <field name="product_uom_qty" readonly="1"/>
-                            <field name="qty_delivered" readonly="1"/>
-                            <field name="qty_invoiced"/>
-                            <field name="product_uom" readonly="1"/>
+                            <field name="product_uom" invisible="1"/>
+                            <field name="product_uom_category_id" invisible="1"/>
+                            <field name="product_uom_readonly" invisible="1"/>
+                            <field name="qty_delivered_method" invisible="1"/>
+                            <field name="state" invisible="1"/>
+                            <field name="product_id"
+                                context="{
+                                    'partner_id': order_partner_id,
+                                    'quantity': product_uom_qty,
+                                    'uom': product_uom,
+                                    'company_id': company_id,
+                                }"
```

Comment:
Not sure this context is still necessary (and planning to someday remove it from the existing view), do you know what/where it's used ?

Reply:
I couldn't see where it is used, but I was afraid of breaking something by removing it

Reply:
AFAIK, it was mostly for some magical contextual price logic that has been since removed.